pointerevent_touch-action-keyboard WPT test is failing

lilyspiniolas opened the pull request:
#### af121ecc9ac7dd766a449872dc6e349dbb2c68bf
<pre>
pointerevent_touch-action-keyboard WPT test is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=282651">https://bugs.webkit.org/show_bug.cgi?id=282651</a>
<a href="https://rdar.apple.com/139317785">rdar://139317785</a>

Reviewed by Abrar Rahman Protyasha.

The test now waits for the keyup event to fire for the arrow_down key before sending the
arrow_right key press to avoid flakiness. Assertions now only print once instead of
potentially printing for every scroll event after a key press. Added explicit assertions
for horizontal scrolling and for vertical scrolling for clarity rather than asserting both
in a single statement, and made both check that the scrolling occurred in the correct
direction.

* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard.html:

Canonical link: <a href="https://commits.webkit.org/286317@main">https://commits.webkit.org/286317@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/760669e9e35a816d61dfb97938f716b94929f0ef

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75632 "Passed style check") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-18-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80116 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/26896 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2920 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/59314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/26896 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78699 "Passed tests") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64960 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/39678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22446 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25225 "Built successfully") | 
| | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22783 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81591 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/2971 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/1872 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/67548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3122 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64936 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66850 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16660 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/10809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/8956 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2928 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/2953 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/3888 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/2960 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard.html:
```diff
@@ -78,66 +78,66 @@ <h4 id="desc">Test Description: Press DOWN ARROW key. Wait for description updat
-                // Inject keyboard scroll inputs.
-                const arrow_down = "\uE015";
-                const arrow_right = "\uE014";
+                document.addEventListener('keyup' ,async function(e) {
```
Spacing
```suggestion
                document.addEventListener('keyup', async function(e) {
```

lilyspiniolas committed:
pointerevent_touch-action-keyboard WPT test is failing
https://bugs.webkit.org/show_bug.cgi?id=282651
rdar://139317785

Reviewed by Abrar Rahman Protyasha.

The test now waits for the keyup event to fire for the arrow_down key before sending the
arrow_right key press to avoid flakiness. Assertions now only print once instead of
potentially printing for every scroll event after a key press. Added explicit assertions
for horizontal scrolling and for vertical scrolling for clarity rather than asserting both
in a single statement, and made both check that the scrolling occurred in the correct
direction.

* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard.html:

Canonical link: https://commits.webkit.org/286317@main
